src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AuditCategory.java

```diff
@@ -0,0 +1,38 @@
+package com.amazon.opendistroforelasticsearch.security.auditlog.impl;
+
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Objects;
+
+public enum AuditCategory {
+    BAD_HEADERS,
+    FAILED_LOGIN,
+    MISSING_PRIVILEGES,
+    GRANTED_PRIVILEGES,
+    OPENDISTRO_SECURITY_INDEX_ATTEMPT,
+    SSL_EXCEPTION,
+    AUTHENTICATED,
+    COMPLIANCE_DOC_READ,
+    COMPLIANCE_DOC_WRITE,
+    COMPLIANCE_EXTERNAL_CONFIG,
+    COMPLIANCE_INTERNAL_CONFIG_READ,
+    COMPLIANCE_INTERNAL_CONFIG_WRITE;
+
+    public static EnumSet<AuditCategory> parse(final List<String> categories) {
+        EnumSet<AuditCategory> set = EnumSet.noneOf(AuditCategory.class);
+        if (categories == null)
+            return set;
+
+        categories
```

Comment:
Can also be written as
```
 return categories.stream()
                .filter(Objects::nonNull)
                .map(category -> AuditCategory.valueOf(category.toUpperCase()))
                .collect(Collectors.toCollection(() -> EnumSet.noneOf(AudityCategory.class)));
```

Reply:
This will throw IllegalArgumentException exception if invalid category string is passed in the List.

Reply:
Won't a try/catch block resolve that?

Reply:
It would quit the looping for remaining categories instead of just ignoring the current one

Reply:
Cool thanks for following up.